src/main/java/lottery/LottoNumbers.java

```diff
@@ -27,11 +27,30 @@ private LottoNumbers(Set<Integer> numbers) {
 		return Collections.unmodifiableSet(numbers);
 	}
 
-	public int matchCountWith(LottoNumbers numbers){
+	public LotteryMatchType getMatchTypeWith(LottoNumbers numbers, int bonus) {
+		LotteryMatchType matchType = LotteryMatchType.fromInteger(getSameNumberCount(numbers));
+		if (isBonusMatchCount(bonus, matchType)) {
+			return LotteryMatchType.FIVE_MATCH_WITH_BONUS;
+		}
+		if (isNotContainsBonus(bonus, matchType)) {
+			return LotteryMatchType.FIVE_MATCH;
+		}
+		return matchType;
+	}
```

Comment:
```
규칙 2: else 예약어를 쓰지 않는다.
```

if를 2개 사용한다고 위 원칙이 지켜진다고 보긴 어려울 것 같습니다.

enum의 values()는 순서를 보장했던 것 같은데 그렇다고 하면 FIVE_MATCH가 우선적으로 반환되어 아래와 같이도 사용할 수 있을 것 같아요.

다만 조금 혼란을 줄 순 있으니 여러 방법들을 한번 고민해보시면 좋을 것 같습니다.


```suggestion
		if (isBonusMatchCount(bonus, matchType)) {
			return LotteryMatchType.FIVE_MATCH_WITH_BONUS;
		}
		return matchType;
	}
```

Reply:
저도 이부분에 대해서 고민이 많았습니다. values()가 순서를 보장하지만, enum의 순서를 개발자가 조금이라도 변경하면- 사이드 이펙트가 발견될 수 있을 확률이 높은 것 같아, 위와같이 수정했습니다. 

조금 더 고민해보겠습니다!